runtime/runtime/src/verifier.rs

```diff
@@ -20,10 +22,98 @@ use near_store::{
 };
 
 use crate::config::{total_prepaid_gas, tx_cost, TransactionCost};
+use crate::near_primitives::account::{Account, FunctionCallPermission};
+use crate::near_primitives::trie_key::trie_key_parsers;
 use crate::VerificationResult;
-use near_primitives::checked_feature;
-use near_primitives::runtime::config::RuntimeConfig;
-use near_primitives::types::BlockHeight;
+use near_primitives::hash::CryptoHash;
+
+/// Checks if given account has enough balance for storage stake, and returns:
+///  - None if account has enough balance or is a zero-balance account
+///  - Some(insufficient_balance) if account doesn't have enough and how much need to be added,
+///  - Err(message) if account has invalid storage usage or amount/locked.
+///
+/// Read details of state staking
+/// <https://nomicon.io/Economics/README.html#state-stake>.
+pub fn get_insufficient_storage_stake(
```

Comment:
the naming of the function is a little bit confusing (as you're trying to 'get' an error condition - insufficient storage stake).

I'd suggest renaming to:

pub fn check_storage_state(...) -> Result<(), StorageStateError>

And then StorageStakeError is an enum:

pub enum StorageStakeError {
    InsufficientBalance(Balance),
    OtherError(String)
}